src/compile.ts

```diff
@@ -80,8 +74,139 @@ export function compile(
         ts.factory.createStringLiteral("functionless")
       );
 
-      const statements = sf.statements.map(
-        (stmt) => visitor(stmt) as ts.Statement
+      const registerName = ts.factory.createUniqueName("register");
+      const bindName = ts.factory.createUniqueName("bind");
+      const globals: (ts.FunctionDeclaration | ts.Statement)[] = [
+        // function register(func, ast) {
+        //   func[Symbol.for("functionless:ast")] = ast;
+        //   return func;
+        // }
+        ts.factory.createFunctionDeclaration(
+          undefined,
+          undefined,
+          undefined,
+          registerName,
+          undefined,
+          [param("func"), param("ast")],
+          undefined,
+          ts.factory.createBlock([
+            // func[Symbol.for("functionless:ast")] = ast;
+            setSymbol("func", ReflectionSymbolNames.AST, "ast"),
+
+            // return func;
+            ret("func"),
+          ])
+        ),
```

Comment:
nit: could these be created in the parent closure like `functionless` to reduce the number of time it is run?